tests/python/unittest/test_gluon_rnn.py

```diff
@@ -890,152 +657,53 @@ def __init__(self, rnn_size, time_step, **kwargs):
+            def forward(self, inputs, valid_len):
                 outputs, states = self.bi_lstm.unroll(self.time_step, inputs, valid_length=valid_len,
                                                       layout='NTC', merge_outputs=True)
                 return outputs, states
+            
+            def infer_shape(self, x, *args):
+                self.bi_lstm.infer_shape(0, x.shape[x.ndim-1], True)
 
         rnn_size = 100
         net = BiLSTM(rnn_size, length)
+        inputs_data = mx.np.random.uniform(size=(10, length, 50))
+        net.infer_shape(inputs_data)
         net.initialize()
         net.hybridize()
-        inputs_data = mx.nd.random.uniform(shape=(10, length, 50))
-        valid_len = mx.nd.array([length]*10)
+        valid_len = mx.np.array([length]*10)
         outputs, _ = net(inputs_data, valid_len)
         assert outputs.shape == (10, length, 200)
 
     _check_bidirectional_unroll_valid_length(1)
     _check_bidirectional_unroll_valid_length(3)
 
 
-def check_rnn_cell(cell, in_shape=(10, 50), out_shape=(10, 100), begin_state=None):
-    inputs = [mx.sym.Variable('rnn_t%d_data'%i) for i in range(3)]
-    outputs, _ = cell.unroll(3, inputs, begin_state=begin_state)
-    outputs = mx.sym.Group(outputs)
-    assert sorted(cell.collect_params().keys()) == ['h2h_bias', 'h2h_weight',
-                                                    'i2h_bias', 'i2h_weight']
-    assert outputs.list_outputs() == [type(cell).__name__.lower() + name for name in ['_t0_out_output', '_t1_out_output', '_t2_out_output']]
-
-    args, outs, auxs = outputs.infer_shape(rnn_t0_data=in_shape,
-                                           rnn_t1_data=in_shape,
-                                           rnn_t2_data=in_shape)
-    assert outs == [out_shape] * 3
-
-
 def check_rnn_forward(layer, inputs):
     inputs.attach_grad()
     layer.initialize()
     with mx.autograd.record():
         layer.unroll(3, inputs, merge_outputs=True)[0].backward()
         mx.autograd.backward(layer.unroll(3, inputs, merge_outputs=False)[0])
-    mx.nd.waitall()
+mx.npx.waitall()
 
 
 def test_rnn_cells():
     check_rnn_forward(gluon.rnn.Conv1DLSTMCell((5, 7), 10, (3,), (3,)),
-                      mx.nd.ones((8, 3, 5, 7)))
+                      mx.np.ones((8, 3, 5, 7)))
     check_rnn_forward(gluon.rnn.Conv1DRNNCell((5, 7), 10, (3,), (3,)),
-                      mx.nd.ones((8, 3, 5, 7)))
+                      mx.np.ones((8, 3, 5, 7)))
     check_rnn_forward(gluon.rnn.Conv1DGRUCell((5, 7), 10, (3,), (3,)),
-                      mx.nd.ones((8, 3, 5, 7)))
+                      mx.np.ones((8, 3, 5, 7)))
 
     net = mx.gluon.rnn.SequentialRNNCell()
     net.add(gluon.rnn.Conv1DLSTMCell((5, 7), 10, (3,), (3,)))
     net.add(gluon.rnn.Conv1DRNNCell((10, 5), 11, (3,), (3,)))
     net.add(gluon.rnn.Conv1DGRUCell((11, 3), 12, (3,), (3,)))
-    check_rnn_forward(net, mx.nd.ones((8, 3, 5, 7)))
-
-
-def test_convrnn():
```

Comment:
Should this test be adapted to testing the front-end infer_shape functions instead?